src/wallet/coinselection.cpp

```diff
@@ -395,3 +382,54 @@ CAmount GetSelectionWaste(const std::set<CInputCoin>& inputs, CAmount change_cos
 
     return waste;
 }
+
+void SelectionResult::ComputeAndSetWaste(CAmount change_cost)
+{
+    m_waste = GetSelectionWaste(m_selected_inputs, change_cost, m_target, m_use_effective);
+}
+
+CAmount SelectionResult::GetWaste() const
+{
+    Assume(m_waste != std::nullopt);
+    return *m_waste;
+}
+
+CAmount SelectionResult::GetSelectedValue() const
+{
+    CAmount ret = 0;
+    for (const auto& coin : m_selected_inputs) {
+        ret += coin.txout.nValue;
+    }
+    return ret;
+}
+
+void SelectionResult::Clear()
+{
+    m_selected_inputs.clear();
+}
```

Comment:
In 1b1ad42
Should this function also clear `m_waste`?

Reply:
Done